app/src/main/java/com/github/multimatum_team/multimatum/CalendarActivity.kt

```diff
@@ -1,54 +1,114 @@
 package com.github.multimatum_team.multimatum
 
+import android.content.Context
+import android.graphics.Rect
 import android.os.Bundle
+import android.view.KeyEvent
+import android.view.MotionEvent
 import android.view.View
+import android.view.inputmethod.InputMethodManager
 import android.widget.CalendarView
+import android.widget.EditText
 import androidx.activity.viewModels
 import androidx.appcompat.app.AppCompatActivity
 import com.github.multimatum_team.multimatum.model.Deadline
 import com.github.multimatum_team.multimatum.model.DeadlineState
-import com.github.multimatum_team.multimatum.repository.DeadlineRepository
 import com.github.multimatum_team.multimatum.viewmodel.DeadlineListViewModel
 import com.google.android.material.textfield.TextInputEditText
 import dagger.hilt.android.AndroidEntryPoint
-import kotlinx.coroutines.runBlocking
 import java.time.Instant
+import java.time.LocalDate
 import java.time.ZoneId
-import javax.inject.Inject
+
+
+//import javax.inject.Inject
 
 @AndroidEntryPoint
 class CalendarActivity : AppCompatActivity() {
-    @Inject
-    lateinit var deadlineRepository: DeadlineRepository
-
     private val viewModel: DeadlineListViewModel by viewModels()
+    private var selectedDate: LocalDate = Instant.now().atZone(ZoneId.systemDefault()).toLocalDate()
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         initCalendar()
+        initTextInput()
     }
 
+    /*
+    This function initialize the calender, it allows the app to listen to
+    an eventual date selection by the user.
+     */
     private fun initCalendar() {
         setContentView(R.layout.activity_calendar)
+
+        // Get the calendar view
+        val calendar = findViewById<CalendarView>(R.id.calendar_view)
+
+        // Update the selected date in case of a selection in the calendar
+        calendar.setOnDateChangeListener { _, year, month, day ->
```

Comment:
It is possible to test this to increase coverage ?

Reply:
Good point. But honestly, I don't know how to test it. The CalendarView is a very complex structure (containing a lot of sub-structures). To test that, we need to be able to simulate a date selection on the calendar, and I don't know how to do it. If someone know how to do that, feel free to tell me.